torch_geometric/nn/conv/message_passing.py

```diff
@@ -480,12 +486,16 @@ def aggregate(self, inputs: Tensor, index: Tensor,
         """
         aggr = self.aggr if aggr is None else aggr
         assert aggr is not None
-        if ptr is not None:
-            ptr = expand_left(ptr, dim=self.node_dim, dims=inputs.dim())
-            return segment_csr(inputs, ptr, reduce=aggr)
+        if isinstance(self.aggr, Aggregation):
+            return aggr(inputs, index=index, ptr=ptr, dim_size=dim_size,
+                        dim=self.node_dim)
         else:
-            return scatter(inputs, index, dim=self.node_dim, dim_size=dim_size,
-                           reduce=aggr)
+            if ptr is not None:
```

Comment:
I don't think we need this conditional here. The goal is to always simply call `return self.aggr(...)`.